File: group#2/김명기/6주차/2.cpp
// 네트워크
#include <bits/stdc++.h>
using namespace std;

int solution(int n, vector<vector<int>> computers)
{
    int answer = 0;
    vector<bool> v(n, false);
    function<void(int)> f = [&](int ix)
    {
        if (v[ix])
            return;
        v[ix] = true;
        for (int i = 0; i < n; i++)
            if (computers[ix][i])
                f(i);
    };
    for (int i = 0; i < n; i++)
        if (!v[i])
        {
            answer++;
            f(i);
        }
    return answer;
}

File: group#2/김명기/9주차/2.py
# 보석 쇼핑

def check(d, gems):
    for gem in gems:
        if d[gem] == 0:
            return False
    return True


def solution(gems):
    answer = []
    d = {}
    m = 987654321
    n = len(gems)
    s = set()
    for gem in gems:
        d[gem] = 0
    a, b = 0, 0
    while b < n:
        d[gems[b]] += 1
        s.add(gems[b])
        b += 1
        while len(s) == len(d):
            if m > b-a:
                m = b-a
                answer = [a+1, b]
            d[gems[a]] -= 1
            a += 1
            if d[gems[a-1]] == 0:
                s.remove(gems[a-1])
                break
    return answer


gems = eval(input())
print(solution(gems))
